### PR TITLE
Check if 'aup' in rciamAttributes is empty

### PR DESCRIPTION
**Pull request by lionick**
1) check if ['rciamAttributes']['aup'] is empty 
2) Remove hover background color when 'Continue' button is disabled

### CODE REVIEW

File: lib/Auth/Process/UpdateAUP.php [hunk 1]
```diff
@@ -63,7 +63,14 @@ public function process(&$state)
...
         // Check if user is in blacklist
-        if(!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) &&  !empty(array_intersect($state['Attributes'][$this->config['userIdAttribute']], $this->config['userIdBlacklist']))) {
+        // or $state['rciamAttributes']['aup'] is empty
+        if ((!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) && !empty(
+            array_intersect(
+              $state['Attributes'][$this->config['userIdAttribute']],
+              $this->config['userIdBlacklist']
+            )
+            ))
+          || empty($state['rciamAttributes']['aup'])) {
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/UpdateAUP.php]: Add debug messages:
1. `"[aup:UpdateAUP] Skipping blacklisted user with id " . $state['Attributes'][$this->config['userIdAttribute']`
2. "[aup:UpdateAUP] No AUP information found in state - skipping"

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
Improve logging

File: lib/Auth/Process/UpdateAUP.php (hunk 1 above)
Comment (lionick) [reply lib/Auth/Process/UpdateAUP.php]: done

File: lib/Auth/Process/UpdateAUP.php [hunk 2]
```diff
@@ -59,15 +59,23 @@ public function __construct($config, $reserved)
...
-            SimpleSAML_Logger::debug("[aup] process: Skipping blacklisted SP ". var_export($state['SPMetadata']['entityid'], true));
+            SimpleSAML_Logger::debug("[aup:UpdateAUP] Skipping blacklisted SP ". var_export($state['SPMetadata']['entityid'], true));
             return;
         }
         // Check if user is in blacklist
-        if(!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) &&  !empty(array_intersect($state['Attributes'][$this->config['userIdAttribute']], $this->config['userIdBlacklist']))) {
+        if ((!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) && !empty(array_intersect(
+            $state['Attributes'][$this->config['userIdAttribute']],
+            $this->config['userIdBlacklist'])))) {
+            SimpleSAML_Logger::debug("[aup:UpdateAUP] Skipping blacklisted user with id " . $state['Attributes'][$this->config['userIdAttribute']]);
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/UpdateAUP.php]: this is an array so better use `var_export($state['Attributes'][$this->config['userIdAttribute'], true)`

File: lib/Auth/Process/UpdateAUP.php [hunk 3]
```diff
@@ -59,15 +59,23 @@ public function __construct($config, $reserved)
...
+            return;
+        }
+        // Check if $state['rciamAttributes']['aup'] is empty
+        if (empty($state['rciamAttributes']['aup'])) {
+            SimpleSAML_Logger::debug("[aup:UpdateAUP] No AUP information found in state - skipping");
             return;
         }
         try {
-            SimpleSAML_Logger::debug('[aup] process: ' . $this->config['userIdAttribute'] . ' ' . var_export($state['Attributes'][$this->config['userIdAttribute']],true));
+            SimpleSAML_Logger::debug('[aup:UpdateAUP] ' . $this->config['userIdAttribute'] . ' ' . var_export($state['Attributes'][$this->config['userIdAttribute']],true));
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/UpdateAUP.php]: should we check if the userIdAttribute exists or this has already been checked?

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: lib/Auth/Process/UpdateAUP.php (hunk 2 above)
Comment (lionick) [reply lib/Auth/Process/UpdateAUP.php]: done

File: lib/Auth/Process/UpdateAUP.php (hunk 3 above)
Comment (lionick) [reply lib/Auth/Process/UpdateAUP.php]: I just removed it as it was only for testing purposes actually.

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**